server/handlers/postSurveys.js

```diff
@@ -0,0 +1,44 @@
+const { getDB } = require('../databaseConnection');
+const addQuestions = require('../queries/addQuestions');
+const createSurvey = require('../queries/createSurvey');
+/* Survey schema:
+    {
+        id: ObjectID (auto generated)
+        title: string
+        description: string
+        status: created/published/closed
+        dateCreated: date
+        dateToPublish: date
+        datePublished: date
+        dateToClose: date
+        dateClosed: date
+        anonymous: boolean
+        recipients: [{employeeId: ObjectId completed: boolean}]
+        questions: [{questionId: ObjectId position: int}]
+        responses: [{
+            employeeId: ObjectID (or null if anon.)
+            answers: [{
+                questionId: ObjectId
+                answer: string/int (depends on question type)
+                comment: null or string
+                }]
+        }]
+    }
+*/
+
+const postSurveys = async (req, res) => {
+  const surveyObj = req.body;
+  const questions = [...surveyObj.questions];
+  const db = getDB();
+
+  try {
+    const questionIds = await addQuestions(
+      questions,
+      db.collection('questions'),
+    );
+    await createSurvey(surveyObj, questionIds, db.collection('surveys'));
```

Comment:
Please call `getDB` in your createSurvey query. Your handler should be as modular as possible. What if we wanted to use a different db in the future? other than Mongo? that way your handler would call the query with those two params: surveyObj and questionIds and it would be independent of where that data would be stored 😄 
to recap: consider deleting line 32, and removing third arg in line 39

Reply:
This has been added to in the next commit